mmdet/core/evaluation/eval_hooks.py

```diff
@@ -70,16 +132,42 @@ def evaluation_flag(self, runner):
     def after_train_epoch(self, runner):
         if not self.evaluation_flag(runner):
             return
+        current_ckpt_path = osp.join(runner.work_dir,
+                                     f'epoch_{runner.epoch + 1}.pth')
+        json_path = osp.join(runner.work_dir, 'best.json')
+
+        if osp.exists(json_path) and len(self.best_json) == 0:
+            self.best_json = mmcv.load(json_path)
+            self.best_score = self.best_json['best_score']
+            self.best_ckpt = self.best_json['best_ckpt']
+            self.key_indicator = self.best_json['key_indicator']
+
         from mmdet.apis import single_gpu_test
         results = single_gpu_test(runner.model, self.dataloader, show=False)
-        self.evaluate(runner, results)
+        key_score = self.evaluate(runner, results)
+        if self.save_best and self.compare_func(key_score, self.best_score):
+            self.best_score = key_score
+            self.logger.info(
+                f'Now best checkpoint is epoch_{runner.epoch + 1}.pth.'
+                f'Best {self.key_indicator} is {self.best_score}')
+            self.best_json['best_score'] = self.best_score
+            self.best_json['best_ckpt'] = current_ckpt_path
+            self.best_json['key_indicator'] = self.key_indicator
+            mmcv.dump(self.best_json, json_path)
```

Comment:
It is better to symlink or copy a `best.pth` instead of saving a json file, just like `latest.pth`.